mlflow/server/handlers.py

```diff
@@ -1230,6 +1230,70 @@ def assert_arg_exists(arg_name, arg):
     return response
 
 
+@catch_mlflow_exception
+def upload_artifact_handler():
+    _logger.info("upload_artifact_handler")
+    args = request.args
+    run_uuid = args.get("run_uuid")
+    if not run_uuid:
+        raise MlflowException(
+            message="Request must specify run_uuid.",
+            error_code=INVALID_PARAMETER_VALUE,
+        )
+    path = args.get("path")
+    if not path:
+        raise MlflowException(
+            message="Request must specify path.",
+            error_code=INVALID_PARAMETER_VALUE,
+        )
+
+    if request.content_length > 10 * 1024 * 1024:
+        raise MlflowException(
+            message="Artifact size is too large. Max size is 10MB.",
+            error_code=INVALID_PARAMETER_VALUE,
+        )
+
+    data = request.data
+
+    run = _get_tracking_store().get_run(run_uuid)
+    artifact_dir = run.info.artifact_uri
+
+    basename = posixpath.basename(path)
+    dirname = posixpath.dirname(path)
+
+    def _log_artifact_to_repo(f, run, dirname, artifact_dir):
+        if _is_servable_proxied_run_artifact_root(run.info.artifact_uri):
+            artifact_repo = _get_artifact_repo_mlflow_artifacts()
+            path_to_log = (
+                os.path.join(run.info.experiment_id, run.info.run_id, "artifacts", dirname)
+                if dirname
+                else os.path.join(run.info.experiment_id, run.info.run_id, "artifacts")
+            )
+        else:
+            artifact_repo = get_artifact_repository(artifact_dir)
+            path_to_log = dirname
+
+        artifact_repo.log_artifact(f.name, path_to_log)
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+        dir_path = os.path.join(tmpdir, dirname) if dirname else tmpdir
+        file_path = os.path.join(dir_path, basename)
+
+        # Ensure directory exists
+        if dirname:
+            os.makedirs(dir_path, exist_ok=True)
+
+        with open(file_path, "wb") as f:
+            f.write(data)
+            f.flush()
+            f.seek(0)
```

Comment:
Curious why we need flush and seek. Can we pass `fiel_path` to `_log_artifact_to_repo`?